src/Test/L0/Worker/SetEnvFileCommandL0.cs

```diff
@@ -250,75 +247,144 @@ public void SetEnvFileCommand_Heredoc_SkipEmptyLines()
                     "EOF",
                     string.Empty,
                 };
-                WriteContent(envFile, content);
+                TestUtil.WriteContent(envFile, content);
                 _setEnvFileCommand.ProcessCommand(_executionContext.Object, envFile, null);
                 Assert.Equal(0, _issues.Count);
                 Assert.Equal(2, _executionContext.Object.Global.EnvironmentVariables.Count);
-                Assert.Equal($"hello{Environment.NewLine}world", _executionContext.Object.Global.EnvironmentVariables["MY_ENV"]);
-                Assert.Equal($"HELLO{Environment.NewLine}AGAIN", _executionContext.Object.Global.EnvironmentVariables["MY_ENV_2"]);
+                Assert.Equal($"hello{BREAK}world", _executionContext.Object.Global.EnvironmentVariables["MY_ENV"]);
+                Assert.Equal($"HELLO{BREAK}AGAIN", _executionContext.Object.Global.EnvironmentVariables["MY_ENV_2"]);
             }
         }
 
         [Fact]
         [Trait("Level", "L0")]
         [Trait("Category", "Worker")]
-        public void SetEnvFileCommand_Heredoc_SpecialCharacters()
+        public void SetEnvFileCommand_Heredoc_EdgeCases()
         {
             using (var hostContext = Setup())
             {
                 var envFile = Path.Combine(_rootDirectory, "heredoc");
                 var content = new List<string>
                 {
-                    "MY_ENV<<=EOF",
-                    "hello",
-                    "one",
-                    "=EOF",
-                    "MY_ENV_2<<<EOF",
```

Comment:
It wasn't clear to me that the `=` on line 272 and the final `<` on line 276 are actually considered part of the "end marker", so I split these two cases out into a separate unit test, `SetEnvFileCommand_Heredoc_EndMarkerVariations` that leverages `[InlineData("=EOF")][InlineData("<EOF")]` to make things clearer.